pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -695,6 +695,30 @@ func (r *Builder) Tasks(vmRef ref.Ref) (list []*plan.Task, err error) {
 	return
 }
 
+func (r *Builder) PreferenceName(vmRef ref.Ref, configMap *core.ConfigMap) (name string, err error) {
+	if configMap == nil {
+		err = liberr.New("can't map the OS, missing config map")
+		return
+	}
+	vm := &model.Workload{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+		return
+	}
+
+	name, ok := configMap.Data[vm.GuestID]
+	if !ok {
+		err = liberr.Wrap(err,
+			"nothing fits the input OS")
```

Comment:
same here

Reply:
added more data to align with the rest of the liberr wrap